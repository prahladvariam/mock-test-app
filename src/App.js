import React, { useState, useEffect } from "react";
import axios from "axios";
import './App.css';

function App() {
  const [users, setUser] = useState([]);
  
  useEffect(() => {
    async function getData() {
      const response = axios
        .get(`https://jsonplaceholder.typicode.com/users`)
        .then((response) => {
          const usersData = response.data
          var result = [];
          var keys = Object.keys(usersData);
          keys.forEach(function(key){
          result.push(usersData[key]);
          setUser(result)
    });
        });   
        
    }
    getData();
  }, [{users}]);

  return (
    <div className="usersDetails">
      <h1>Users Details</h1>
        {
          users.map(user => {
            return(
              <div key={user.id} >
                  <div>{user.id}. {user.name}<br/>{user.email}<br/><br/></div>
              </div>
            )
          })
        }
    </div>
  );
}

export default App;
