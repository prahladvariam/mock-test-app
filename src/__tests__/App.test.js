import App from "../App";
import React, { useEffect, useState } from 'react';
import { shallow } from "enzyme"
import { rest } from "msw" 
import { setupServer } from "msw/node"
import { createWaitForElement } from "enzyme-wait"

const server = setupServer()
describe("Testing .....", () => {
    
    beforeAll(() => {
        server.listen()
    })

    afterEach(() => {
        server.resetHandlers()
    })

    beforeEach(() => {
        shallow(<App />)
    })

    afterAll(() => {
        server.close()
    })

    const mocksetState = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation((initialState) => [initialState, mocksetState]);

    it("component rendering", () => {
        const component = shallow(<App />)
        const listExist = component.find("h1").exists()
        expect(listExist).toBe(true)
        expect(component).toMatchSnapshot()
    });
    
    test("When state data is changed ", () => {
        const component = shallow(<App />)
        expect(mocksetState).not.toHaveBeenCalled();
    })
    
    test("Api working properly", (done) => {
        const waitForData = createWaitForElement("div")
        server.use(rest.get(`https://jsonplaceholder.typicode.com/users/1`, (req, res, ctx) => {
            return res(ctx.json([{ id : 1 , name: "MOck"}])) 
        }))
        const component = shallow(<App />)
        waitForData(component).then((component) => {
            console.log("IN");
            expect(component).toMatchSnapshot()
            done()
        })
    })

})