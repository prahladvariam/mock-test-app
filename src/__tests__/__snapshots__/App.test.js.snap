// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ..... Api working properly 1`] = `
<div
  className="usersDetails"
>
  <h1>
    Users Details
  </h1>
</div>
`;

exports[`Testing ..... component rendering 1`] = `
<div
  className="usersDetails"
>
  <h1>
    Users Details
  </h1>
</div>
`;
